docs/developer-guide.md

```diff
@@ -0,0 +1,84 @@
+
+* Spring-boot 
+* Mybatis
+* PageHelper
+* Swagger2
+* Zuul
+
+The code structure of the backend is shown below.
+
+![backend-code](img/pulsar-manager-backend-code.png)
+
+The backend configuration file is [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/application.properties).
+The database schema is defined [here](https://github.com/streamnative/pulsar-manager/blob/master/src/main/resources/META-INF/sql/mysql-schema.sql)
+Add new RESTful api
+Restful api defines the specification. If the RESTful api needs backend processing, it needs to be prefixed with pulsar-manager. If you want to directly forward the request to the broker, you can directly define it according to the specification of Pulsar RESTful api.
+
+Let's take a new tenant as an example to illustrate how to develop in pulsar manager.
+1. Define a RESTful api
+Restful api that requires backend processing
+List all tenants
+GET /pulsar-manager/admin/v2/tenants
+     (2) Restful api directly forwarded to a broker
+Add a new tenant
+PUT /admin/v2/tenants/{tenantName}
+
+2. Define a service interface.
+Service is used to implement business logic and add interface.
+Here, the `TenantsService` interface is added under the service folder.
+
+![tenants-service](img/tenants-service.png)
+
+3. Implement a service interface.
+Here, some aggregation is done for the data queried from broker.
+
+![tenants-service-impl](img/tenants-service-impl.png)
+
+4. Add a test a test under the test folder, mock data and verify results.
+
+![tenants-service-impl-test](img/tenants-service-impl-test.png)
+
+5. Develop RESTful features.
+Add a `TenantsController` file under the controller folder to add an api to obtain the tenant list.
+In the following class you can add some description information, error response, etc.
+
+![tenants-controller](img/tenants-controller.png)
+
+## Frontend
+
+The frontend is developed based on [vue-element-admi](https://panjiachen.github.io/vue-element-admin-site/guide/) and [Element](https://element.eleme.cn/#/en-US).
+The development of the frontend is [here](https://github.com/streamnative/pulsar-manager/tree/master/front-end/src/views/management).
+
+![frontend-code](img/frontend-code.png)
+
+The frontend code structure is shown as below.
+* api: code of RESTful API icons: code of icons
+* router: code of frontend routings
+* lang: code of multiple languages
+* views: code of front pages
+
+### Develop a page
+
+1. Add a RESTful API interface.
+The RESTful interface is defined below the folder api. 
+
+![tenants-api](img/tenants-api.png)
+
+
+2. Add a new page.
+Add a new page [here](https://github.com/streamnative/pulsar-manager/tree/master/front-end/src/views/management). 
+For example, [this](https://github.com/streamnative/pulsar-manager/blob/master/front-end/src/views/management/tenants/index.vue) is a tenant page. )
+
+3. Localization
+You can add  multiple languages to the corresponding configuration folder under the `lang` folder.
```

Comment:
the same layout issue

![image](https://user-images.githubusercontent.com/50226895/64433574-57c90b80-d0f1-11e9-94fe-83c9d1caf9b7.png)

Reply:
Any thoughts on this layout issue?